packages/node-build-scripts/sass/sassVariableParser.mjs

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ */
+
+// @ts-check
+
+import postcss from "postcss";
+import scssParser from "postcss-scss";
+import simpleVars from "postcss-simple-vars";
+
+export const DEFAULT_FLAG = "!default";
+
+/**
+ * Parses Sass source file contents for variables using "postcss-simple-vars"
```

Comment:
on second thought, we might be able to stick with [get-sass-vars](https://github.com/niksy/get-sass-vars) instead of switching to this custom implementation. I think at some point during the long history of this PR, I got convinced that we had to migrate away from that library because of our custom functions, but I think it can continue to work for us. and it seems to be maintained well enough (last update was Jan 2023).